api/trace/api.go

```diff
@@ -24,9 +24,30 @@ import (
 )
 
 type Provider interface {
-	// Tracer creates a named tracer that implements Tracer interface.
-	// If the name is an empty string then provider uses default name.
-	Tracer(name string) Tracer
+	// Tracer creates a tracer that implements the Tracer interface.
+	// The instrumentationName must be name of the library providing
+	// instrumentation. This name may be the same as the instrumented code
+	// only if that code as built-in instrumentation. If the instrumentation
+	// is empty, then the will assign a default name.
+	Tracer(instrumentationName string, opts ...TracerOption) Tracer
+}
+
+// TODO (MrAlias): unify this API option design:
+// https://github.com/open-telemetry/opentelemetry-go/issues/536
+
+// TracerConfig contains options for a Tracer.
+type TracerConfig struct {
+	InstrumentationVersion string
```

Comment:
Do you set the default value anywhere? Should probably the Go SDK version.

Reply:
I was thinking the default should be empty, in which case no version attribute would be added to the Resource. (Does the specification say?)

Reply:
Looks like specification doesn't say anything about that, so empty string should be fine.

Reply:
I think the default version in the sense that if no version is specified but a valid instrumentation library name is given should indeed be empty.

However, I think I'm going to update to include a "default" version if the instrumentation library name is not provided or is invalid. That seems reasonable enough given we are setting the name there.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/802/commits/4f25d4abf88f50fd08202b59a59f538c1de08c9f